### PR TITLE
resolved_ts: track 1PC

### LINKED ISSUE #13353: resolved_ts:  IngestSST do not update the tracked_index
### Is your feature request related to a problem? Please describe:

In [`group_row_changes`](https://github.com/tikv/tikv/blob/0030aeb90a840140a935fbb0181b6a62b5e680b3/components/resolved_ts/src/cmd.rs#L193), only update tracked_index of the `CmdType::Put `and `CmdType::Delete`, tracked_index not updated in IngestSST and [1PC](https://github.com/tikv/tikv/blob/6dcd0b3d66945583a133e60838533a5d12b3a487/components/resolved_ts/src/endpoint.rs#L193) scenarios.

TiFlash needs to use safe_ts to determine whether the peer's data synchronization progress is far behind the leader, so if the tracked_index is not updated in IngestSST scenario, the result is inaccurate. (ref: https://github.com/pingcap/tiflash/issues/4902)

### Describe the feature you'd like:

Update tracked_index in `IngestSST` and 1PC scenarios.

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>

### What is changed and how it works?

Issue Number: Close #13353

What's Changed:

```
Update tracked_index in 1PC scenario. If 1PC is not tracked, it may affect the accuracy of stale read after 1PC DML.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
Update tracked_index in 1PC scenario.
```

### CODE REVIEW

File: components/raftstore/src/store/util.rs [hunk 1]
```diff
@@ -871,7 +871,7 @@ impl RegionReadProgressRegistry {
             .lock()
             .unwrap()
             .get(region_id)
-            .map(|rp| rp.core.lock().unwrap().applied_index)
```
Comment (5kbpers) [inline components/raftstore/src/store/util.rs:874]: Why change this?

Comment (hehechen) [reply components/raftstore/src/store/util.rs:874]: This method is just used in UT if we need to get tracked_index.
The tracked_index is [`read_state.idx`](https://github.com/tikv/tikv/blob/1a9446f334c2d29417f656fa8ef4ec6a1dda95f7/components/raftstore/src/store/util.rs#L1260) but not `applied_index`, so I change this.

**Review by 5kbpers — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**